model/dependencies.go

```diff
@@ -115,8 +117,10 @@ func (di *dependencyIncluder) handle(pair TVPair, activationInfo *specificActiva
 	// also mark this newly generated dependency as inactive.
 	pairSpecifiesActivation := activationInfo.taskOrVariantHasSpecificActivation(pair.Variant, pair.TaskName)
 	for _, dep := range deps {
-		di.updateDeactivationMap(activationInfo, dep, generatedVariants, pairSpecifiesActivation)
-		ok, err := di.handle(dep, activationInfo, generatedVariants)
+		if isRoot {
```

Comment:
we love a basecase! We may also want to comment here about why we only want to do this for the root case though